### PR TITLE
fix graphsync

### PR DESCRIPTION
**Pull request by turuslan**
- `StreamReadBuffer` yamux workaround, used in graphsync.
- `StreamProxy` wrapper base.
- refactor graphsync from `PeerId + optional<Multiaddress>` to `PeerInfo`.
- retrieval client "accepted" check.
- storage client retry polling on error.
- longer storage client polling interval.
- fix `MpoolPushMessage`.
- `uptime` metric in seconds.
- paych `addFunds` deadlock.
- fix typos.

### CODE REVIEW

File: core/common/libp2p/stream_proxy.hpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <libp2p/connection/stream.hpp>
+
+namespace libp2p::connection {
+  struct StreamProxy : Stream {
```
Comment (ortyomka) [inline core/common/libp2p/stream_proxy.hpp:11]: `StreamProxy` is the same with `Stream`. remove?

File: core/common/libp2p/stream_read_buffer.hpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+#include "common/bytes.hpp"
+#include "common/libp2p/stream_proxy.hpp"
+#include "common/outcome.hpp"
+
+namespace libp2p::connection {
+  using fc::Bytes;
+  using fc::BytesOut;
+
+  struct StreamReadBuffer : StreamProxy,
```
Comment (ortyomka) [inline core/common/libp2p/stream_read_buffer.hpp:16]:
```suggestion
  struct StreamReadBuffer : Stream,
```

File: core/common/libp2p/stream_proxy.hpp (hunk 1 above)
Comment (turuslan) [reply core/common/libp2p/stream_proxy.hpp:11]: > `StreamProxy` _wrapper base_
without this each wrapper implementation needs to implement all methods, even if it wants to override just one

File: core/common/libp2p/stream_proxy.hpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    std::shared_ptr<Stream> stream;
```
Comment (ortyomka) [inline core/common/libp2p/stream_proxy.hpp:12]: Should it be public?

File: core/node/main/builder.cpp [hunk 4]
```diff
@@ -288,7 +288,7 @@ namespace fc::node {
             std::make_shared<PieceIOImpl>("/tmp/fuhon/piece_io"));
     // timer is set to 100 ms
```
Comment (ortyomka) [inline core/node/main/builder.cpp:289]:
```suggestion
    // timer is set to 5000 ms
```

**Review by ortyomka — APPROVED** [review]

File: core/common/libp2p/stream_proxy.hpp (hunk 3 above)
Comment (turuslan) [reply core/common/libp2p/stream_proxy.hpp:12]: subclasses need it

File: test/core/vm/actor/builtin/types/miner/monies_test_v2.cpp [hunk 5]
```diff
@@ -50,7 +50,7 @@ namespace fc::vm::actor::builtin::v2::miner {
 
   TEST_F(MoniesTestv2, TestPledgePenaltyForTermination) {
     const TokenAmount initial_pledge = TokenAmount{1 << 10};
-    day_reward = initial_pledge, big_initial_pledge_factor;
+    day_reward = bigdiv(initial_pledge, big_initial_pledge_factor);
```
Comment (Alexey-N-Chernyshov) [inline test/core/vm/actor/builtin/types/miner/monies_test_v2.cpp:53]: 👍

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**